### PR TITLE
Improving library.

### PR DESCRIPTION
**Pull request by NachtRaveVL**
Improving library. Renaming some methods to more closely match other Signal libraries. Adding a detach method. Improving const correctness.

### CODE REVIEW

File: library.properties [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-version=1.0
+version=1.1
 author=Tom Stewart <tomstewart89@hotmail.com>
-maintainer=Tom Stewart <tomstewart89@hotmail.com>
+maintainer=NachtRaveVL <nachtravevl@gmail.com>
 sentence=A simple implementation of Signals and Slots for writing more flexible callbacks
 paragraph=Similar to Boost::Signals and a handy algorithm when writing event driven programs
 category=Other
-url=https://github.com/tomstewart89/Callback
+url=https://github.com/NachtRaveVL/Callback
```
Comment (tomstewart89) [inline library.properties]: I appreciate the work you put in on this PR, but as the original author of the library I'd prefer for the url to continue to point to my repo. I hope I'm not being too selfish here

Comment (NachtRaveVL) [reply library.properties]: Oh not at all selfish! :p Go ahead and merge it into your stuff! ^_^

**Comment by NachtRaveVL** [comment]
No need for a gentle ping! lol I've been very busy with work lately and so haven't gotten back to this.

Please by all means modify the files as you see fit and merge them in, it's by no means an issue. =)

**Comment by tomstewart89** [comment]
Great! Thanks again for the PR!

**Pull request merged**